cmd/utils/nodecmd/nodeflags.go

```diff
@@ -21,353 +21,432 @@
+}
+
+func KsenAppFlags() []cli.Flag {
+	flags := append([]cli.Flag{}, KsenNodeFlags()...)
+	flags = append(flags, CommonRPCFlags...)
+	flags = append(flags, ConsoleFlags...)
+	flags = append(flags, debug.Flags...)
+	return flags
+}
+
 // Common flags that configure the node
 var CommonNodeFlags = []cli.Flag{
-	utils.BootnodesFlag,
-	utils.IdentityFlag,
-	utils.UnlockedAccountFlag,
-	utils.PasswordFileFlag,
-	utils.DbTypeFlag,
-	utils.DataDirFlag,
-	utils.OverwriteGenesisFlag,
-	utils.StartBlockNumberFlag,
-	utils.KeyStoreDirFlag,
-	utils.TxPoolNoLocalsFlag,
-	utils.TxPoolAllowLocalAnchorTxFlag,
-	utils.TxPoolDenyRemoteTxFlag,
-	utils.TxPoolJournalFlag,
-	utils.TxPoolJournalIntervalFlag,
-	utils.TxPoolPriceLimitFlag,
-	utils.TxPoolPriceBumpFlag,
-	utils.TxPoolExecSlotsAccountFlag,
-	utils.TxPoolExecSlotsAllFlag,
-	utils.TxPoolNonExecSlotsAccountFlag,
-	utils.TxPoolNonExecSlotsAllFlag,
-	utils.TxPoolLifetimeFlag,
-	utils.TxPoolKeepLocalsFlag,
-	utils.SyncModeFlag,
-	utils.GCModeFlag,
-	utils.LightKDFFlag,
-	utils.SingleDBFlag,
-	utils.NumStateTrieShardsFlag,
-	utils.LevelDBCompressionTypeFlag,
-	utils.LevelDBNoBufferPoolFlag,
-	utils.DBNoPerformanceMetricsFlag,
-	utils.DynamoDBTableNameFlag,
-	utils.DynamoDBRegionFlag,
-	utils.DynamoDBIsProvisionedFlag,
-	utils.DynamoDBReadCapacityFlag,
-	utils.DynamoDBWriteCapacityFlag,
-	utils.DynamoDBReadOnlyFlag,
-	utils.LevelDBCacheSizeFlag,
-	utils.NoParallelDBWriteFlag,
-	utils.SenderTxHashIndexingFlag,
-	utils.TrieMemoryCacheSizeFlag,
-	utils.TrieBlockIntervalFlag,
-	utils.TriesInMemoryFlag,
-	utils.CacheTypeFlag,
-	utils.CacheScaleFlag,
-	utils.CacheUsageLevelFlag,
-	utils.MemorySizeFlag,
-	utils.TrieNodeCacheTypeFlag,
-	utils.NumFetcherPrefetchWorkerFlag,
-	utils.UseSnapshotForPrefetchFlag,
-	utils.TrieNodeCacheLimitFlag,
-	utils.TrieNodeCacheSavePeriodFlag,
-	utils.TrieNodeCacheRedisEndpointsFlag,
-	utils.TrieNodeCacheRedisClusterFlag,
-	utils.TrieNodeCacheRedisPublishBlockFlag,
-	utils.TrieNodeCacheRedisSubscribeBlockFlag,
-	utils.ListenPortFlag,
-	utils.SubListenPortFlag,
-	utils.MultiChannelUseFlag,
-	utils.MaxConnectionsFlag,
-	utils.MaxRequestContentLengthFlag,
-	utils.MaxPendingPeersFlag,
-	utils.TargetGasLimitFlag,
-	utils.NATFlag,
-	utils.NoDiscoverFlag,
-	utils.RWTimerWaitTimeFlag,
-	utils.RWTimerIntervalFlag,
-	utils.NetrestrictFlag,
-	utils.NodeKeyFileFlag,
-	utils.NodeKeyHexFlag,
-	utils.VMEnableDebugFlag,
-	utils.VMLogTargetFlag,
-	utils.VMTraceInternalTxFlag,
-	utils.NetworkIdFlag,
-	utils.RPCCORSDomainFlag,
-	utils.RPCVirtualHostsFlag,
-	utils.RPCNonEthCompatibleFlag,
-	utils.MetricsEnabledFlag,
-	utils.PrometheusExporterFlag,
-	utils.PrometheusExporterPortFlag,
-	utils.ExtraDataFlag,
-	utils.SrvTypeFlag,
-	utils.AutoRestartFlag,
-	utils.RestartTimeOutFlag,
-	utils.DaemonPathFlag,
-	utils.ConfigFileFlag,
-	utils.APIFilterGetLogsMaxItemsFlag,
-	utils.APIFilterGetLogsDeadlineFlag,
-	utils.OpcodeComputationCostLimitFlag,
-	utils.SnapshotFlag,
-	utils.SnapshotCacheSizeFlag,
+	utils.ConfFlag,
```

Comment:
Don't we need to call `altsrc.NewStringFlag(utils.ConfFlag)`?

Also I couldn't find ConfFlag.

Reply:
`altsrc.NewStringFlag(utils.ConfFlag)` is for reading the flag in yaml file.
`--conf` flag option is a special case we have to put in command line like `./kcn --conf conf.yaml`